functions/_pure_prompt_k8s.fish

```diff
@@ -4,14 +4,13 @@ function _pure_prompt_k8s
         and _pure_check_availability pure_enable_k8s kubectl
         and test -n (_pure_k8s_context) # todo: use $(cmd) syntax when Fish 3.3.1 is dropped
 
-        set --local symbol (_pure_set_color $pure_color_k8s_prefix)$pure_symbol_k8s_prefix
         set --local context (_pure_set_color $pure_color_k8s_context)(_pure_k8s_context)
-        set --local namespace (_pure_set_color $pure_color_k8s_namespace)(_pure_k8s_namespace)
 
-        if test -n "$namespace"
-            set --local namespace (_pure_set_color $pure_color_k8s_namespace)default
-        end
```

Comment:
@bobsoppe I move this logic in `functions/_pure_k8s_namespace.fish` also changed the flag to `-z` as I reckon you want the `default` value when no namespace is set.

Fish doc [Operators for text strings](https://fishshell.com/docs/current/cmds/test.html#operators-for-text-strings)
```
 -n STRING
    Returns true if the length of STRING is non-zero.

-z STRING
    Returns true if the length of STRING is zero.

```

Reply:
![image](https://github.com/pure-fish/pure/assets/1212392/b9f88252-4e35-4262-a027-dba6381249dc)